lemur/tests/js_test.js

```diff
@@ -0,0 +1,66 @@
+// This file consists of unit tests for functions in utility.js
+QUnit.module( 'group student_data_entry' );
+QUnit.test('isInArray', function(assert) {
+    assert.deepEqual(isInArray('1', ['1','2','3']), true);
+    assert.deepEqual(isInArray('', ['1','2','3']), false);
+});
+QUnit.test('isNumeric', function(assert) {
+    assert.deepEqual(isNumeric('-1.5'), true);
+    assert.deepEqual(isNumeric('-1.5a'), false); 
+});
+QUnit.test('checkInputStudent', function(assert) {
+    function numberEntry(observation, valueRange, expected) {
+        assert.deepEqual(checkInputStudent(observation, 'Number', valueRange, '')['wrongInput'], !expected);
+    }
+    function textEntry(observation, valueCandidates, expected) {
+        assert.deepEqual(checkInputStudent(observation, 'Text', '', valueCandidates)['wrongInput'], !expected);
+    }
+    numberEntry('1.5','1.0-10.0',true);
+    numberEntry('0.8','1.0-10.0',false);
+    textEntry('Critical','Critical,Helpful',true);
+    textEntry('Criticall','Critical,Helpful',false);
+});
+
+QUnit.module( 'group student_select_lab' );
+QUnit.test('decomposeClassId', function(assert) {
+    var classId = 'BIOL101_FALL2016';
+    assert.deepEqual(decomposeClassId(classId), ['BIOL101','FALL2016']);
+    assert.notDeepEqual(decomposeClassId(classId), undefined);
```

Comment:
these statements are redundant with your previous one. You are guaranteed that it is not undefined because you know it is `['BIOL101','FALL2016']`